samples/matter/common/src/diagnostic/log_backend_diagnostic.cpp

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "diagnostic_logs_provider.h"
+
+#include <zephyr/logging/log.h>
+#include <zephyr/logging/log_backend.h>
+
+static uint32_t sLogFormat = LOG_OUTPUT_TEXT;
+/* The buffer size was selected to be able to store single log line (usually in Matter we have logs < 200 B).
+ * In case the log will not fit it will be printed anyway, but usng fragmentation, so setting a small buffer
+ * increases the time of printing.
+ */
+static uint8_t sLogBuffer[256];
+
+static int RedirectNetworkLog(uint8_t *data, size_t length, void *context)
+{
+	Nrf::Matter::DiagnosticLogProvider::GetInstance().PushLog(
+		chip::app::Clusters::DiagnosticLogs::IntentEnum::kNetworkDiag, data, length);
+
+	return length;
+}
+
+static int RedirectEndUserLog(uint8_t *data, size_t length, void *context)
+{
+	Nrf::Matter::DiagnosticLogProvider::GetInstance().PushLog(
+		chip::app::Clusters::DiagnosticLogs::IntentEnum::kEndUserSupport, data, length);
+
+	return length;
+}
+
+LOG_OUTPUT_DEFINE(log_output_network, RedirectNetworkLog, sLogBuffer, sizeof(sLogBuffer));
+LOG_OUTPUT_DEFINE(log_output_end_user, RedirectEndUserLog, sLogBuffer, sizeof(sLogBuffer));
```

Comment:
Is it ok that we use the same buffer for both log output backends?

Reply:
I guess, it is OK because logs are added sequentially, but I wonder if there is a possibility to corrupt this buffer when it is filled by two sources.

Reply:
I don't think it's possible, because as you said these calls are used sequentially.